app/lib/substrate.py

```diff
@@ -164,16 +164,39 @@ def substrate_query_url(url, module, function,params=[]):
 def substrate_sudo_relay_xcm_call(para_id, encoded_message):
     substrate_client = get_relay_chain_client()
     keypair = Keypair.create_from_seed(network_sudo_seed())
-    return substrate_xcm_call(substrate_client, keypair, para_id, encoded_message)
+    return substrate_xcm_sudo_transact_call(substrate_client, keypair, para_id, encoded_message)
 
 
-def substrate_xcm_call(substrate_client, keypair, para_id, encoded_message):
+def substrate_xcm_sudo_transact_call(substrate_client, keypair, para_id, encoded_message):
     payload = substrate_client.compose_call(
         call_module='XcmPallet',
         call_function='send',
         call_params={
-            'dest': {'V0': {'X1': {'Parachain': para_id}}},
-            'message': {'V0': {'Transact': ('Superuser', 1000000000, {'encoded': str(encoded_message)})}}
-        }
-    )
+            'dest': {
+                'V3': {
+                    "parents": 0,
+                    'interior': {
+                        'X1': {
+                            'Parachain': para_id
+                        }
+                    }
+                }
+            },
+            'message': {
+                'V3': [[
```

Comment:
Note: for some calls, you may need to add an explicit `UnpaidExecution` instruction:
https://github.com/paritytech/devops/issues/2480#issuecomment-1491885756

I am waiting for developers' comments on how to detect which relay/para chain needs `UnpaidExecution`

Reply:
I will add UnpaidExecution. It looks like it is required on newer Cumulus runtimes. Here is the latest call that was successful on Westend:

```
 poetry run python scripts/get-extrinsic-call.py wss://westend-rpc.polkadot.io 0xfb74949c613397ceb4966b89a7627e8e1f850326020f1f7953db84238392cb7f 0x748b8fa1a94f5d2ae08a6091420c5c2a75d302416fc6f4d24d1a10f72ae2a9fb
{
    'call_index': '0x1500',
    'call_function': 'sudo',
    'call_module': 'Sudo',
    'call_args': [
        {
            'name': 'call',
            'type': 'RuntimeCall',
            'value': {
                'call_index': '0x6300',
                'call_function': 'send',
                'call_module': 'XcmPallet',
                'call_args': [
                    {
                        'name': 'dest',
                        'type': 'VersionedMultiLocation',
                        'value': {
                            'V3': {
                                'parents': 0,
                                'interior': {
                                    'X1': {
                                        'Parachain': 1002
                                    }
                                }
                            }
                        }
                    },
                    {
                        'name': 'message',
                        'type': 'VersionedXcm<()>',
                        'value': {
                            'V3': [
                                {
                                    'UnpaidExecution': {
                                        'weight_limit': 'Unlimited',
                                        'check_origin': null
                                    }
                                },
                                {
                                    'Transact': {
                                        'origin_kind': 'Superuser',
                                        'require_weight_at_most': {
                                            'ref_time': 1000000000,
                                            'proof_size': 1024
                                        },
                                        'call': {
                                            'encoded': '0x0102a98a60a9b40a0a2fed51b611a8ba078f64d9501f7fa68130e05df711e81301f3'
                                        }
                                    }
                                }
                            ]
                        }
                    }
                ],
                'call_hash': '0xeb3fc6df14508bc631abd458b8d73beb461a879575ba6db66a01f904d7594623'
            }
        }
    ],
    'call_hash': '0x0bc7b8df49e62ba7d3da382cc0b7770492e92649331024c9c11eaf171b90969f'
}
```